Return-Path: <linux-arm-kernel-bounces+thomas.petazzoni=free-electrons.com@lists.infradead.org>
Delivered-To: thomas.petazzoni@free-electrons.com
Received: by mail.free-electrons.com (Postfix, from userid 106)
	id E8A021AE; Wed,  4 Apr 2012 20:37:20 +0200 (CEST)
X-Spam-Checker-Version: SpamAssassin 3.3.1 (2010-03-16) on
	mail.free-electrons.com
X-Spam-Level: 
X-Spam-Status: No, score=-5.6 required=1.0 tests=AWL,BAYES_00,
	RCVD_IN_DNSWL_MED,T_RP_MATCHES_RCVD shortcircuit=no autolearn=ham
	version=3.3.1
Received: from merlin.infradead.org (merlin.infradead.org [205.233.59.134])
	by mail.free-electrons.com (Postfix) with ESMTPS id 675E912A
	for <thomas.petazzoni@free-electrons.com>; Wed,  4 Apr 2012 20:37:10 +0200 (CEST)
Received: from localhost ([::1] helo=merlin.infradead.org)
	by merlin.infradead.org with esmtp (Exim 4.76 #1 (Red Hat Linux))
	id 1SFV56-0000So-HZ; Wed, 04 Apr 2012 18:37:20 +0000
Received: from mail-wg0-f49.google.com ([74.125.82.49])
 by merlin.infradead.org with esmtps (Exim 4.76 #1 (Red Hat Linux))
 id 1SFV4z-0000SE-Vl
 for linux-arm-kernel@lists.infradead.org; Wed, 04 Apr 2012 18:37:14 +0000
Received: by wgbdr1 with SMTP id dr1so360083wgb.18
 for <linux-arm-kernel@lists.infradead.org>;
 Wed, 04 Apr 2012 11:37:11 -0700 (PDT)
X-Google-DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
 d=google.com; s=20120113;
 h=from:to:cc:subject:date:message-id:x-mailer:x-gm-message-state;
 bh=mdT8MBLbu5vw3Ql0o0yVv1r64XWPIi+xvTrNokfOiWA=;
 b=IV+U/1wZPNM0KJBnMjpm8MCBSR9A9ecoEROQocd6NIBvYGLJzFNma80F7DxMxpYzWt
 xMBzsTddxDv4glHCZ0s+2mauK1n3ckO3fb6BBTet315X0Jj+HHH31/2qSBdDO7V1ZJBW
 flOKTi75/BJIeeqHQbFOHCfsD6ewRF3gFDUoZLRzFjqPTgjKPbttJICIJOsp8mndR9ju
 CqheaFFl0p+whKwnD1st7Rs/CIaSM6ZSpK+Ts8AOhKtAHWnPNcQAgubILz0dj27bIzf+
 KBq4pFDk0vzX5ZeM2cw2gCkUHRtF5Ro50Q6qmH7QmmUeTbJns2oMi8hVucxoLKY3t7x9
 sduA==
Received: by 10.180.91.10 with SMTP id ca10mr7711191wib.17.1333564631490;
 Wed, 04 Apr 2012 11:37:11 -0700 (PDT)
Received: from localhost.localdomain ([188.77.211.255])
 by mx.google.com with ESMTPS id 17sm7060020wis.0.2012.04.04.11.37.09
 (version=TLSv1/SSLv3 cipher=OTHER);
 Wed, 04 Apr 2012 11:37:10 -0700 (PDT)
From: Javier Martinez Canillas <javier@dowhile0.org>
To: Tony Lindgren <tony@atomide.com>
Subject: [RESEND PATCH 1/2] ARM: OMAP2+: nand: Make board_onenand_init()
 visible to board code
Date: Wed,  4 Apr 2012 20:36:25 +0200
Message-Id: <1333564586-3259-1-git-send-email-javier@dowhile0.org>
X-Mailer: git-send-email 1.7.7.6
X-Gm-Message-State: ALoCoQmIk8aNDreTL/tiYHZ34Wduf7Br1GODPnocJLk8FAD3+IJjWdVzHn0i7wAfChtd6IEQ5nbj
Cc: linaro-kernel@lists.linaro.org, linaro-dev@lists.linaro.org,
 patches@linaro.org,
 =?UTF-8?q?Enric=20Balletb=C3=B2=20i=20Serra?= <eballetbo@gmail.com>,
 Russell King <linux@arm.linux.org.uk>,
 Javier Martinez Canillas <javier@dowhile0.org>, linux-omap@vger.kernel.org,
 linux-arm-kernel@lists.infradead.org
X-BeenThere: linux-arm-kernel@lists.infradead.org
X-Mailman-Version: 2.1.14
Precedence: list
List-Id: <linux-arm-kernel.lists.infradead.org>
List-Unsubscribe: <http://lists.infradead.org/mailman/options/linux-arm-kernel>, 
 <mailto:linux-arm-kernel-request@lists.infradead.org?subject=unsubscribe>
List-Archive: <http://lists.infradead.org/pipermail/linux-arm-kernel/>
List-Post: <mailto:linux-arm-kernel@lists.infradead.org>
List-Help: <mailto:linux-arm-kernel-request@lists.infradead.org?subject=help>
List-Subscribe: <http://lists.infradead.org/mailman/listinfo/linux-arm-kernel>, 
 <mailto:linux-arm-kernel-request@lists.infradead.org?subject=subscribe>
MIME-Version: 1.0
Content-Type: text/plain; charset="us-ascii"
Content-Transfer-Encoding: 7bit
Sender: linux-arm-kernel-bounces@lists.infradead.org
Errors-To: linux-arm-kernel-bounces+thomas.petazzoni=free-electrons.com@lists.infradead.org

board_onenand_init() and board_nand_init() initialization functions are
used to initialize OneNAND and NAND memories respectively. But only
board_nand_init() was visible to be used from board code. This patch makes
possible to initialize a OneNAND flash memory within platform code.

Signed-off-by: Javier Martinez Canillas <javier@dowhile0.org>
---
 arch/arm/mach-omap2/board-flash.c |    4 ++--
 arch/arm/mach-omap2/board-flash.h |   11 +++++++++++
 2 files changed, 13 insertions(+), 2 deletions(-)

diff --git a/arch/arm/mach-omap2/board-flash.c b/arch/arm/mach-omap2/board-flash.c
index 0349fd2..70a81f9 100644
--- a/arch/arm/mach-omap2/board-flash.c
+++ b/arch/arm/mach-omap2/board-flash.c
@@ -87,7 +87,7 @@ static struct omap_onenand_platform_data board_onenand_data = {
 	.dma_channel	= -1,   /* disable DMA in OMAP OneNAND driver */
 };
 
-static void
+void
 __init board_onenand_init(struct mtd_partition *onenand_parts,
 				u8 nr_parts, u8 cs)
 {
@@ -98,7 +98,7 @@ __init board_onenand_init(struct mtd_partition *onenand_parts,
 	gpmc_onenand_init(&board_onenand_data);
 }
 #else
-static void
+void
 __init board_onenand_init(struct mtd_partition *nor_parts, u8 nr_parts, u8 cs)
 {
 }
diff --git a/arch/arm/mach-omap2/board-flash.h b/arch/arm/mach-omap2/board-flash.h
index d25503a..c44b70d 100644
--- a/arch/arm/mach-omap2/board-flash.h
+++ b/arch/arm/mach-omap2/board-flash.h
@@ -47,3 +47,14 @@ static inline void board_nand_init(struct mtd_partition *nand_parts,
 {
 }
 #endif
+
+#if defined(CONFIG_MTD_ONENAND_OMAP2) || \
+		defined(CONFIG_MTD_ONENAND_OMAP2_MODULE)
+extern void board_onenand_init(struct mtd_partition *nand_parts,
+					u8 nr_parts, u8 cs);
+#else
+static inline void board_onenand_init(struct mtd_partition *nand_parts,
+					u8 nr_parts, u8 cs)
+{
+}
+#endif
-- 
1.7.7.6


_______________________________________________
linux-arm-kernel mailing list
linux-arm-kernel@lists.infradead.org
http://lists.infradead.org/mailman/listinfo/linux-arm-kernel
